src/server/pfs/cmds/cmds.go

```diff
@@ -687,14 +687,7 @@ Any pachctl command that can take a Commit ID, can take a branch name instead.`,
 			}
 			var headCommit *pfs.Commit
 			if head != "" {
-				if uuid.IsUUIDWithoutDashes(head) {
-					headCommit = branch.Repo.NewCommit("", head)
-				} else {
-					headCommit, err = cmdutil.ParseCommit(head)
```

Comment:
I think we still need something to handle this case, because you may have multiple commits matching the given commitset in a repo (one per branch).  In that case, the RPC would fail unless the user provides a fully-qualified commit (i.e. `repo@branch=id`).  I realize the repo is redundant in this case, but the branch is important and I don't think it's worth inventing a new syntax.